docs/extending/query.md

```diff
@@ -1,197 +1,209 @@
+		],
+	],
+],
 ```
 
-### get_image_url(): string|null
-
-By default, the `get_image_url` method proxies to the `get_image_url` method of the query's data source. Override this method to provide an image URL that will represent the query in the UI.
-
-### get_request_method(): string
+### request_method: string
 
-By default, `get_request_method` returns `'GET'`. Override this method if your query uses a different HTTP request method.
+The `request_method` property defines the HTTP request method used by the query. By default, it is `'GET'`.
 
-### get_request_headers( array $input_variables ): array
+### request_headers: array|callable
 
-By default, the `get_request_headers` method proxies to the `get_request_headers` method of the query's data source. Override this method to provide custom request headers for the query. The input variables for the current request are provided as an associative array (`[ $var_name => $value ]`).
+The `request_headers` property defines the request headers for the query. It can be an associative array or a callable function that returns an associative array. The callable function accepts an associative array of input variables (`[ $var_name => $value ]`). If omitted, the query will use the request headers defined by the data source.
 
 ### Example
 
 ```php
-public function get_request_headers( array $input_variables ): array|WP_Error {
+'request_headers' => function( array $input_variables ) use ( $data_source ): array {
 	return array_merge(
-		$this->get_data_source()->get_request_headers(),
-		[ 'X-Product-ID' => $input_variables['product_id'] ]
+		$data_source->get_request_headers(),
+		[ 'X-Foo' => $input_variables['foo'] ]
 	);
-}
+},
 ```
 
-### get_request_body( array $input_variables ): array|null
+### request_body: array|callable
 
-Override this method to define a request body for this query. The return value will be converted to JSON using `wp_json_encode`. The input variables for the current request are provided as an associative array (`[ $var_name => $value ]`).
+The `request_body` property defines the request body for the query. It can be an associative array or a callable function that returns an associative array. The callable function accepts an associative array of input variables (`[ $var_name => $value ]`). If omitted, the query will not have a request body.
 
-### get_query_name(): string
+### cache_ttl: int|null|callable
 
-Override this method to specify a name that represents the query in UI.
+The `cache_ttl` property defines how long the query response should be cached, in seconds. It can be an integer, a callable function that returns an integer, or `null`. The callable function accepts an associative array of input variables (`[ $var_name => $value ]`).
 
-### get_query_runner(): QueryRunnerInterface
+A value of `-1` indicates the query should not be cached. A value of `null` indicates the default TTL should be used (60 seconds). If omitted, the default TTL is used.
```

Comment:
Can we add a bit of information here about what is required for this caching to work? It appears that internally this TTL is passed to `wp_cache_set()`/`wp_cache_get()`. How about:

```suggestion
A value of `-1` indicates the query should not be cached. A value of `null` indicates the default TTL should be used (60 seconds). If omitted, the default TTL is used.

Remote data blocks utilize the WordPress object cache (`wp_cache_get()` / `wp_cache_set()`) for response caching. Ensure a persistent object cache plugin is provided by your platform or installed for this value to be respected.
```